pinot-core/src/main/java/com/linkedin/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -141,13 +141,17 @@ private boolean isFitForMetadataBasedPlan(BrokerRequest brokerRequest) {
     }
 
     List<AggregationInfo> aggregationsInfo = brokerRequest.getAggregationsInfo();
-    if (aggregationsInfo != null && aggregationsInfo.size() == 1 && !brokerRequest.isSetGroupBy()) {
-      if (aggregationsInfo.get(0)
-          .getAggregationType()
-          .equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.COUNT.getName())) {
-        return true;
+    if (aggregationsInfo != null && !brokerRequest.isSetGroupBy()) {
+      for (int i = 0; i < aggregationsInfo.size(); i++) {
+        String aggregationType = aggregationsInfo.get(i).getAggregationType();
```

Comment:
just checking for function name is not sufficient. We also need to know if the column has dictionary. Also, we are not really using metadata to answer these queries, so we should have separate method to check if the query can dictionary.